### ISSUE TITLE
'spacy_transformers' has no attribute 'TransformersLanguage'

### ISSUE THREAD

**Issue by mgbvox** [explosion/spacy-transformers#177]
Hi all,

**tl;dr:** spacy is giving me the error 'AttributeError: partially initialized module 'spacy_transformers' has no attribute 'TransformersLanguage' (most likely due to a circular import)
', and it's killing my kernels. Not fun.

**Full breakdown:**

This is a fascinating project, and I'm champing at the bit to use it. But I appear to have hit a roadblock in installation.

I'm using a conda env, characterized by the following .yml:
`name: nlp
channels:
  - marta-sd
  - defaults
  - conda-forge
dependencies:
  - appnope=0.1.0=py38_0
  - attrs=19.3.0=py_0
  - backcall=0.1.0=py38_0
  - blas=1.0=mkl
  - bleach=3.1.4=py_0
  - ca-certificates=2020.1.1=0
  - catalogue=1.0.0=py_0
  - certifi=2020.4.5.1=py38_0
  - cffi=1.14.0=py38hb5b8e2f_0
  - chardet=3.0.4=py38_1003
  - cryptography=2.9.2=py38ha12b0ac_0
  - cycler=0.10.0=py38_0
  - cymem=2.0.3=py38hc84c608_1
  - cython-blis=0.4.1=py38h64e0658_1
  - dbus=1.13.14=h517e14e_0
  - decorator=4.4.2=py_0
  - defusedxml=0.6.0=py_0
  - entrypoints=0.3=py38_0
  - expat=2.2.6=h0a44026_0
  - freetype=2.9.1=hb4e5f40_0
  - gettext=0.19.8.1=h15daf44_3
  - glib=2.63.1=hd977a24_0
  - icu=58.2=h0a44026_3
  - idna=2.9=py_1
  - importlib-metadata=1.6.0=py38h32f6830_0
  - importlib_metadata=1.5.0=py38_0
  - intel-openmp=2020.1=216
  - ipykernel=5.1.4=py38h39e3cac_0
  - ipython=7.13.0=py38h5ca1d4c_0
  - ipython_genutils=0.2.0=py38_0
  - ipywidgets=7.5.1=py_0
  - jedi=0.17.0=py38_0
  - jinja2=2.11.2=py_0
  - joblib=0.14.1=py_0
  - jpeg=9b=he5867d9_2
  - json5=0.9.4=py_0
  - jsonschema=3.2.0=py38_0
  - jupyter=1.0.0=py38_7
  - jupyter_client=6.1.3=py_0
  - jupyter_console=6.1.0=py_0
  - jupyter_core=4.6.3=py38_0
  - jupyterlab=1.2.6=pyhf63ae98_0
  - jupyterlab_server=1.1.1=py_0
  - kiwisolver=1.2.0=py38h04f5b5a_0
  - libcxx=10.0.0=h1af66ff_2
  - libcxxabi=4.0.1=hcfea43d_1
  - libedit=3.1.20181209=hb402a30_0
  - libffi=3.2.1=h0a44026_6
  - libgfortran=3.0.1=h93005f0_2
  - libiconv=1.16=h1de35cc_0
  - libpng=1.6.37=ha441bb4_0
  - libsodium=1.0.16=h3efe00b_0
  - llvm-openmp=4.0.1=hcfea43d_1
  - markupsafe=1.1.1=py38h1de35cc_0
  - matplotlib=3.1.3=py38_0
  - matplotlib-base=3.1.3=py38h9aa3819_0
  - mistune=0.8.4=py38h1de35cc_1000
  - mkl=2019.4=233
  - mkl-service=2.3.0=py38hfbe908c_0
  - mkl_fft=1.0.15=py38h5e564d8_0
  - mkl_random=1.1.0=py38h6440ff4_0
  - murmurhash=1.0.0=py38hc84c608_0
  - nbconvert=5.6.1=py38_0
  - nbformat=5.0.6=py_0
  - ncurses=6.2=h0a44026_1
  - notebook=6.0.3=py38_0
  - numpy=1.18.1=py38h7241aed_0
  - numpy-base=1.18.1=py38h6575580_1
  - openssl=1.1.1g=h1de35cc_0
  - pandoc=2.2.3.2=0
  - pandocfilters=1.4.2=py38_1
  - parso=0.7.0=py_0
  - pcre=8.43=h0a44026_0
  - pexpect=4.8.0=py38_0
  - pickleshare=0.7.5=py38_1000
  - pip=20.0.2=py38_3
  - plac=0.9.6=py_1
  - preshed=3.0.2=py38hc84c608_2
  - prometheus_client=0.7.1=py_0
  - prompt-toolkit=3.0.4=py_0
  - prompt_toolkit=3.0.4=0
  - ptyprocess=0.6.0=py38_0
  - pycparser=2.20=py_0
  - pygments=2.6.1=py_0
  - pyopenssl=19.1.0=py38_0
  - pyparsing=2.4.7=py_0
  - pyqt=5.9.2=py38h655552a_2
  - pyrsistent=0.16.0=py38h1de35cc_0
  - pysocks=1.7.1=py38_0
  - python=3.8.2=hc70fcce_0
  - python-dateutil=2.8.1=py_0
  - python_abi=3.8=1_cp38
  - pyzmq=18.1.1=py38h0a44026_0
  - qt=5.9.7=h468cd18_1
  - qtconsole=4.7.4=py_0
  - qtpy=1.9.0=py_0
  - readline=8.0=h1de35cc_0
  - requests=2.23.0=py38_0
  - scikit-learn=0.22.1=py38h27c97d8_0
  - scipy=1.4.1=py38h44e99c9_0
  - send2trash=1.5.0=py38_0
  - setuptools=46.2.0=py38_0
  - sip=4.19.8=py38h0a44026_0
  - six=1.14.0=py38_0
  - spacy=2.2.4=py38ha0d09dd_1
  - spacy-lookups-data=0.2.0=py_1
  - sqlite=3.31.1=h5c1f38d_1
  - srsly=1.0.2=py38hc84c608_0
  - terminado=0.8.3=py38_0
  - testpath=0.4.4=py_0
  - thinc=7.4.0=py38ha0d09dd_2
  - tk=8.6.8=ha441bb4_0
  - tornado=6.0.4=py38h1de35cc_1
  - tqdm=4.46.0=pyh9f0ad1d_0
  - traitlets=4.3.3=py38_0
  - urllib3=1.25.8=py38_0
  - wasabi=0.6.0=py_0
  - wcwidth=0.1.9=py_0
  - webencodings=0.5.1=py38_1
  - wheel=0.34.2=py38_0
  - widgetsnbextension=3.5.1=py38_0
  - xz=5.2.5=h1de35cc_0
  - zeromq=4.3.1=h0a44026_3
  - zipp=3.1.0=py_0
  - zlib=1.2.11=h1de35cc_3
  - pip:
    - boto3==1.13.9
    - botocore==1.16.9
    - click==7.1.2
    - docutils==0.15.2
    - en-core-web-sm==2.2.5
    - en-trf-bertbaseuncased-lg==2.2.0
    - filelock==3.0.12
    - ftfy==5.7
    - future==0.18.2
    - jmespath==0.10.0
    - regex==2020.5.14
    - s3transfer==0.3.3
    - sacremoses==0.0.43
    - sentencepiece==0.1.90
    - spacy-transformers==0.5.1
    - tokenizers==0.7.0
    - torch==1.5.0
    - torchcontrib==0.0.2
    - transformers==2.0.0
prefix: /Users/mgbvox/opt/anaconda3/envs/nlp`

After installing the en_trf_bertbaseuncased_lg model using:
`python -m spacy download en_trf_bertbaseuncased_lg`

And running the following code in Pycharm:
`import spacy

nlp = spacy.load('en_trf_bertbaseuncased_lg')

doc = nlp("Apple shares rose on the news. Apple pie is delicious.")

print(doc)`

I get the following error:
`AttributeError: partially initialized module 'spacy_transformers' has no attribute 'TransformersLanguage' (most likely due to a circular import)
`

Full traceback:
`Traceback (most recent call last):
  File "/Users/mgbvox/Desktop/CS/Portfolio_and_Co/in_progress/NLP/spacy_transformers.py", line 3, in <module>
    nlp = spacy.load('en_trf_bertbaseuncased_lg')
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/__init__.py", line 30, in load
    return util.load_model(name, **overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 164, in load_model
    return load_model_from_package(name, **overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 185, in load_model_from_package
    return cls.load(**overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/en_trf_bertbaseuncased_lg/__init__.py", line 12, in load
    return load_model_from_init_py(__file__, **overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 228, in load_model_from_init_py
    return load_model_from_path(data_path, meta, **overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 196, in load_model_from_path
    cls = get_lang_class(lang)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 70, in get_lang_class
    if lang in registry.languages:
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/catalogue.py", line 56, in __contains__
    has_entry_point = self.entry_points and self.get_entry_point(name)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/catalogue.py", line 140, in get_entry_point
    return entry_point.load()
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/mgbvox/Desktop/CS/Portfolio_and_Co/in_progress/NLP/spacy_transformers.py", line 3, in <module>
    nlp = spacy.load('en_trf_bertbaseuncased_lg')
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/__init__.py", line 30, in load
    return util.load_model(name, **overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 164, in load_model
    return load_model_from_package(name, **overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 185, in load_model_from_package
    return cls.load(**overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/en_trf_bertbaseuncased_lg/__init__.py", line 12, in load
    return load_model_from_init_py(__file__, **overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 228, in load_model_from_init_py
    return load_model_from_path(data_path, meta, **overrides)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 196, in load_model_from_path
    cls = get_lang_class(lang)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/spacy/util.py", line 70, in get_lang_class
    if lang in registry.languages:
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/catalogue.py", line 56, in __contains__
    has_entry_point = self.entry_points and self.get_entry_point(name)
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/site-packages/catalogue.py", line 140, in get_entry_point
    return entry_point.load()
  File "/Users/mgbvox/opt/anaconda3/envs/nlp/lib/python3.8/importlib/metadata.py", line 79, in load
    return functools.reduce(getattr, attrs, module)
AttributeError: partially initialized module 'spacy_transformers' has no attribute 'TransformersLanguage' (most likely due to a circular import)

Process finished with exit code 1
`

I'm running a 2015 MacBook Pro on the latest stable version of Catalina.

**Comment by adrianeboyd** [comment]
This is not an uncommon problem related to how python searches for imports: you need to rename your top-level script to something other than `spacy_transformers.py`, since it's trying to import that instead of the actual library.

**Comment by mgbvox** [comment]
@adrianeboyd , thanks for the tip! I renamed the script to 'nlp_with_spacy.py'. Now when I run it, I get:

`Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)`

I've never encountered this sort of error before - or at least, not in my day-to-day data engineering work. Thoughts?

**Comment by adrianeboyd** [comment]
Hmm, it's hard to know what is causing that, but a guess might be `torch`? Try installing `torch` according to the instructions on their website in a new virtual environment (preferably installing with the conda instructions if you're using a conda env) and then installing `spacy-transformers` again: https://pytorch.org/get-started/locally/
